frb_example/pure_dart/rust/src/api.rs

```diff
@@ -1005,43 +1029,82 @@ pub fn run_enum_opaque(opaque: EnumOpaque) -> String {
     }
 }
 
-pub fn run_opaque(opaque: Opaque<HideData>) -> String {
+pub fn run_opaque(opaque: RustOpaque<HideData>) -> String {
     opaque.hide_data()
 }
 
-pub fn run_opaque_with_delay(opaque: Opaque<HideData>) -> String {
+pub fn run_opaque_with_delay(opaque: RustOpaque<HideData>) -> String {
     sleep(Duration::from_millis(1000));
     opaque.hide_data()
 }
 
-pub fn opaque_array() -> [Opaque<HideData>; 2] {
-    [Opaque::new(HideData::new()), Opaque::new(HideData::new())]
+pub fn opaque_array() -> [RustOpaque<HideData>; 2] {
+    [
+        RustOpaque::new(HideData::new()),
+        RustOpaque::new(HideData::new()),
+    ]
 }
 
-pub fn opaque_array_run(data: [Opaque<HideData>; 2]) {
+pub fn opaque_array_run(data: [RustOpaque<HideData>; 2]) {
     for i in data {
         i.hide_data();
     }
 }
 
-pub fn opaque_vec() -> Vec<Opaque<HideData>> {
-    vec![Opaque::new(HideData::new()), Opaque::new(HideData::new())]
+pub fn opaque_vec() -> Vec<RustOpaque<HideData>> {
+    vec![
+        RustOpaque::new(HideData::new()),
+        RustOpaque::new(HideData::new()),
+    ]
 }
 
-pub fn opaque_vec_run(data: Vec<Opaque<HideData>>) {
+pub fn opaque_vec_run(data: Vec<RustOpaque<HideData>>) {
     for i in data {
         i.hide_data();
     }
 }
 
 pub fn create_nested_opaque() -> OpaqueNested {
     OpaqueNested {
-        first: Opaque::new(HideData::new()),
-        second: Opaque::new(HideData::new()),
+        first: RustOpaque::new(HideData::new()),
+        second: RustOpaque::new(HideData::new()),
     }
 }
 
 pub fn run_nested_opaque(opaque: OpaqueNested) {
     opaque.first.hide_data();
     opaque.second.hide_data();
 }
+
+pub fn unwrap_rust_opaque(opaque: RustOpaque<HideData>) -> String {
+    let hide_data = opaque.try_unwrap().unwrap();
+    hide_data.hide_data()
+}
+
+pub fn unwrap_dart_opaque(opaque: DartOpaque) -> SyncReturn<String> {
+    #[cfg(target_family = "wasm")]
+    {
+        let js = opaque.try_unwrap().unwrap();
+        SyncReturn(js.as_string().unwrap())
+    }
+    #[cfg(not(target_family = "wasm"))]
+    {
+        let handle = opaque.try_unwrap().unwrap();
+        unsafe { Dart_DeletePersistentHandle_DL_Trampolined(handle) };
```

Comment:
Oops, we should never let users of frb touch these dangerous functions. (this api.rs is an example, so it is like user code not library code)

Reply:
how else to test? without it there will be a leak

Reply:
The main idea is: This pure_dart example is the role model for every frb user. So, do we expect frb users to write down these two lines of code (surely not!).

We may wrap it to some safe operations. Not very sure what is the feature here that you want to expose to the users?

Reply:
The idea is to get the raw Dart_Handle to use in some other Dart_Handle api.

We can make it `unsafe` for native platform and remove it from the example

Reply:
I see. 

uber-nit: Shall we name it like `Box::into_raw`, since I do not expect "unwrap" to suddenly give me a raw data that I have to manage by myself (otherwise leak memory). Feel free to keep the name if the "unwrap" is seen somewhere else in std lib with similar scenario.

It seems to can be safe (instead of unsafe), since `into_raw` is safe as well (it just leaks memory if forget to dispose, so Rust style thinks it is safe).

I am thinking about `Dart_DeletePersistentHandle_DL_Trampolined`. Is it possible we mimic Box, i.e. only expose `DartOpaque::from_raw` function, and do not expose `Dart_DeletePersistentHandle_DL_Trampolined`. Then, when user wants to delete the handle, they just `DartOpaque::from_raw`, and let DartOpaque object be dropped, then DeletePersistHandle is auto called inside. This is just like how we treat raw pointers: We do `Box::from_raw(raw_pointer)` when we want to drop it. Of course this is just one way, feel free to discuss if you have other ways that is commonly used in Rust!

Reply:
```
#[derive(Debug)]
pub struct DartHandleWrap(Option<Dart_PersistentHandle>);

impl DartHandleWrap {
    pub fn into_raw(mut self) -> Dart_PersistentHandle {
        self.0.take().unwrap()
    }
}

impl Drop for DartHandleWrap {
    fn drop(&mut self) {
        if let Some(inner) = self.0 {
            unsafe { Dart_DeletePersistentHandle_DL_Trampolined(inner) }
        }
    }
}
```

Reply:
LGTM with nits: What about adding a simple line of comment mentioning "Option" is used for dropping